planner/core/preprocess.go

```diff
@@ -1653,7 +1653,7 @@ func (p *preprocessor) updateStateFromStaleReadProcessor() error {
 				if err := txnManager.EnterNewTxn(context.TODO(), newTxnRequest); err != nil {
 					return err
 				}
-				if err := txnManager.OnStmtStart(context.TODO()); err != nil {
+				if err := txnManager.OnStmtStart(context.TODO(), nil); err != nil {
```

Comment:
We can add a `GetCurrentStmt()` method to `TxnManager` , so this line can be `txnManager.OnStmtStart(context.TODO(), txnManager.GetCurrentStmt())`

Reply:
It's done.